tests/unit/api/c/merger/test_dmx_merger.cpp

```diff
@@ -0,0 +1,1087 @@
+#include "fff.h"
+
+#if SACN_DYNAMIC_MEM
+#define TestDmxMerger TestDmxMergerDynamic
+#else
+#define TestDmxMerger TestDmxMergerStatic
+#endif
+
+constexpr uint16_t VALID_UNIVERSE_ID = 1;
+constexpr uint16_t INVALID_UNIVERSE_ID = 0;
+constexpr uint8_t VALID_PRIORITY = 100;
+constexpr uint8_t INVALID_PRIORITY = 201;
+
+class TestDmxMerger : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    etcpal_reset_all_fakes();
+    sacn_common_reset_all_fakes();
+    sacn_data_loss_reset_all_fakes();
+    sacn_sockets_reset_all_fakes();
+
+    ASSERT_EQ(sacn_mem_init(1), kEtcPalErrOk);
+    ASSERT_EQ(sacn_dmx_merger_init(), kEtcPalErrOk);
+
+    header_default_.cid = etcpal::Uuid::V4().get();
+    memset(header_default_.source_name, '\0', SACN_SOURCE_NAME_MAX_LEN);
+    header_default_.universe_id = VALID_UNIVERSE_ID;
+    header_default_.priority = VALID_PRIORITY;
+    header_default_.preview = false;
+    header_default_.start_code = 0x00;
+    header_default_.slot_count = DMX_ADDRESS_COUNT;
+    memset(pdata_default_, 0, DMX_ADDRESS_COUNT);
+
+    merger_config_ = SACN_DMX_MERGER_CONFIG_INIT;
+    merger_config_.slots = slots_;
+    merger_config_.slot_owners = slot_owners_;
+    merger_config_.source_count_max = SACN_RECEIVER_INFINITE_SOURCES;
+
+    const char* ns_str = "1234567890abcdef";
+    memcpy(namespace_uuid_.data, ns_str, ETCPAL_UUID_BYTES);
+
+    for (int i = 0; i < DMX_ADDRESS_COUNT; ++i)
+    {
+      test_values_ascending_[i] = i % 256;
+      test_values_descending_[i] = 255 - test_values_ascending_[i];
+    }
+  }
+
+  void TearDown() override
+  {
+    sacn_dmx_merger_deinit();
+    sacn_mem_deinit();
+  }
+
+  void GenV5(int iteration, EtcPalUuid* uuid)
+  {
+    char name[80];
+    sprintf_s(name, "%d", iteration);
+
+    etcpal_generate_v5_uuid(&namespace_uuid_, name, 80, uuid);
+  }
+
+  EtcPalUuid GenV5(int iteration)
+  {
+    EtcPalUuid result;
+    GenV5(iteration, &result);
+    return result;
+  }
+
+  // This determines what kind of merge test TestMerge does.
+  enum class MergeTestType
+  {
+    kUpdateSourceData,      // Merge using sacn_dmx_merger_update_source_data.
+    kUpdateSourceFromSacn,  // Merge using sacn_dmx_merger_update_source_from_sacn.
+    kStopSourcePap          // Merge using sacn_dmx_merger_update_source_data, then call
+                            // sacn_dmx_merger_stop_source_per_address_priority on the second source.
+  };
+
+  void TestMerge(uint8_t priority_1, const uint8_t* values_1, uint16_t values_1_count,
+                 const uint8_t* address_priorities_1, uint16_t address_priorities_1_count, uint8_t priority_2,
+                 const uint8_t* values_2, uint16_t values_2_count, const uint8_t* address_priorities_2,
+                 uint16_t address_priorities_2_count, MergeTestType merge_type)
+  {
+    // Initialize the merger and sources.
+    sacn_source_id_t source_1;
+    sacn_source_id_t source_2;
+    EtcPalUuid source_1_cid = GenV5(1);
```

Comment:
I took another look at this file because @nwagner 's comment brought me back here. I think you can simplify this without needing the GenV5() functions. How about:

```cpp
#include "etcpal/cpp/uuid.h"
// ...
auto source_1_cid = etcpal::Uuid::V5(namespace_uuid_, std::to_string(1));
auto source_2_cid = etcpal::Uuid::V5(namespace_uuid_, std::to_string(2));
// ...
EXPECT_EQ(sacn_dmx_merger_add_source(merger_handle_, &source_1_cid.get(), &source_1), kEtcPalErrOk);
```